main.tf

```diff
@@ -47,69 +47,65 @@ module "key_protect" {
 # Key Protect Key Rings
 ##############################################################################
 
-# Create Key Rings included in var.existing_key_map
+locals {
+  key_rings = flatten([
+    for key_ring in var.key_map :
+    key_ring.existing_key_ring ? [] : [{
+      key_ring_name = key_ring.key_ring_name
+      force_delete  = key_ring.force_delete
+    }]
+  ])
+}
+
 module "key_protect_key_rings" {
   source        = "git::https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-key-ring.git?ref=v2.3.1"
-  for_each      = var.key_map
-  instance_id   = local.key_protect_guid
+  for_each      = { for obj in local.key_rings : obj.key_ring_name => obj }
+  instance_id   = local.kms_guid
   endpoint_type = var.key_ring_endpoint_type
-  key_ring_id   = each.key
-  force_delete  = var.force_delete_key_ring
+  key_ring_id   = each.value.key_ring_name
+  force_delete  = each.value.force_delete
 }
 
 ##############################################################################
 # Key Protect Keys
 ##############################################################################
 
 locals {
-  # These two maps transform the variable data from
-  # {key_ring_1 = [key_1, key_2], key_ring_2 = [key_3, key_4]}
-  # into a more usable data structure of
-  # [{key_name = key_1, key_ring_name = key_ring_1}, {key_name = key_2, key_ring_name = key_ring_1}...]
-
-  existing_key_ring_key_map = flatten([
-    for key_ring, key_list in var.existing_key_map : [
-      for key_name in key_list : {
-        key_name      = key_name
-        key_ring_name = key_ring
-      }
+  key_ring_key_list = flatten([
+    for key_ring in var.key_map : key_ring.existing_key_ring ? [] : [
+      for key_obj in key_ring.keys : merge({
+        key_ring_name = key_ring.key_ring_name
+      }, key_obj)
     ]
   ])
 
-  key_ring_key_map = flatten([
-    for key_ring, key_list in var.key_map : [
-      for key_name in key_list : {
-        key_name      = key_name
-        key_ring_name = key_ring
-      }
+  existing_key_ring_key_list = flatten([
+    for key_ring in var.key_map : !key_ring.existing_key_ring ? [] : [
+      for key_obj in key_ring.keys : merge({
+        key_ring_name = key_ring.key_ring_name
+      }, key_obj)
     ]
   ])
 }
 
 # Create Key Rings and Keys
 module "key_protect_keys" {
-  # depends_on necessary since we're not directly referencing the previously provisioned key rings in the module
-  depends_on = [
-    module.key_protect_key_rings
-  ]
-  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-key.git?ref=v1.2.1"
-  # This for_each is needed to assign a name to the maps in the array so they can be referenced/saved in the terraform graph
-  for_each        = { for map_name in local.key_ring_key_map : "${map_name.key_ring_name}.${map_name.key_name}" => map_name }
+  source          = "git::https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-key.git?ref=v1.2.1"
```

Comment:
can we pull from hashicorp instead? Also maybe rename the module ("key_protect_keys" -> "kms_keys") since its not just for key protect (not sure, that may need a moved block to prevent breaking change)